src/lib/apparmor/genprof.rb

```diff
@@ -26,11 +26,17 @@ def execute
       return nil if @program.nil?
       super
 
+      # TRANSLATORS: file path
+      if @status then
```

Comment:
this does not work in ruby. as Integer is always succeed. Try it yourself in irb ( for simplicity backticks is used instead of popen ).
```
irb(main):005:0> `false`; $?.exitstatus ? "success" : "failed"
=> "success"
irb(main):006:0> `true`; $?.exitstatus ? "success" : "failed"
=> "success"
```

Also ruby does not use then.
So it should look like `if @status == 0` or store boolean in status above and ideally call it `@succeed`.

But to be honest my preferred solution is to simply return boolean in execute method. So it will look like

```
def execute
  ...
  $?.exitstatus == 0
end
....
# and here
if super
  msg = _("Error: Profile for %s not generated due to error in executing aa-genprof") % @program
else
  msg = _("Profile for %s generated.") % @program
end
```

This way you do not depend on any variable. But any working solution is better than current one which does not work.